src/pybind/mgr/crash/module.py

```diff
@@ -50,65 +129,110 @@ def timestamp_filter(self, f):
         :returns: crash reports for which f(time) returns true
         """
         def inner(pair):
-            _, meta = pair
-            meta = json.loads(meta)
-            time = self.time_from_string(meta["timestamp"])
+            _, crash = pair
+            time = self.time_from_string(crash["timestamp"])
             return f(time)
-        matches = filter(inner, six.iteritems(
-            self.get_store_prefix("crash/")))
-        return [(k, json.loads(m)) for k, m in matches]
+        return filter(inner, self.crashes.items())
 
     # command handlers
 
     def do_info(self, cmd, inbuf):
         crashid = cmd['id']
-        key = 'crash/%s' % crashid
-        val = self.get_store(key)
-        if not val:
+        crash = self.crashes.get(crashid)
+        if not crash:
             return errno.EINVAL, '', 'crash info: %s not found' % crashid
+        val = json.dumps(crash, indent=4)
         return 0, val, ''
 
     def do_post(self, cmd, inbuf):
         try:
             metadata = self.validate_crash_metadata(inbuf)
         except Exception as e:
             return errno.EINVAL, '', 'malformed crash metadata: %s' % e
-
         crashid = metadata['crash_id']
-        key = 'crash/%s' % crashid
-        # repeated stores of same item are ignored silently
-        if not self.get_store(key):
-            self.set_store(key, inbuf)
+
+        if not crashid in self.crashes:
```

Comment:
this seems like it is checking the membership of `crashid` in `self.crashes` so it should be:

```suggestion
        if crashid not in self.crashes:
```